ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -104,32 +99,24 @@ pub fn compute_final_sum_share<F: PrimeField, const λ: usize, const P: usize>(z
 ///
 /// The function uses streams since stream offers a chunk method.
 fn recurse_u_or_v<'a, F: PrimeField, J, const λ: usize>(
-    u_or_v_stream: J,
+    u_or_v_iterator: J,
     lagrange_table: &'a LagrangeTable<F, λ, 1>,
-) -> impl Stream<Item = F> + 'a
+) -> impl Iterator<Item = F> + 'a
 where
-    J: Stream<Item = F> + 'a,
+    J: Iterator<Item = F> + 'a,
```

Comment:
generally speaking, taking `IntoInterator` makes your function more generic as it accepts more arguments. If you take `Iterator`, you have to convert vectors and arrays using `.into_iter`, but if you take `IntoIterator` you can accept all iterators as before but also vectors and other collections that implement it with no extra method calls on the callee side.